app/src/main/java/xterminators/spellingbee/gui/GuiController.java

```diff
@@ -339,6 +361,62 @@ private void guessWordButtonClick(ActionEvent e) {
         tbGuess.setText("");
     }
 
+    //private void randomPuzzleButtonClick(ActionEvent e) {
+        
+    //}
+
+    /**
+     * The function that is called whenever the savepuzzlebutton is clicked.
+     * @param e - The button click
+     * @throws IOException - If an I/O error occurs.
+     */
+    private void savePuzzleButtonClick(ActionEvent e) throws IOException{
+        try{
+            // Causes a pop up informing the user that their file was saved and where.
+            showMessage(guiFunctions.savePuzzle() + " in the current directory : "
+             + Paths.get("").toAbsolutePath().toString());
+        }
+        // Catches I/O errors
+        catch ( IOException a){
+            System.out.println("There was an I/O error, please try again.");
```

Comment:
Again here, generally, I would show an error dialog, as the user probably won't see the command line error messages.